src/NuGet.Services.AzureSearch/Analysis/PackageIdCustomTokenizer.cs

```diff
@@ -12,6 +12,6 @@ public static class PackageIdCustomTokenizer
 
         public static readonly PatternTokenizer Instance = new PatternTokenizer(
             Name,
-            @"[.\-_,;:'*#!~+()\[\]{}\s]");
+            @"((?<=[A-Z])(?=[A-Z][a-z]))|([.\-_,;:'*#!~+()\[\]{}\s])");
```

Comment:
This splits on:

1. Whitespace
1. The characters `.`, `\`, `-`, `_`, `,`, `;`, `:`, `'`, `*`, `#`, `!`, `~`, `+`, `(`, `)`, `[`, `]`, `{`, `}`
1. After the first character on patterns like `AAa`. For example, `FOOBar` becomes `FOO` and `Bar`